helix-view/src/editor.rs

```diff
@@ -1828,8 +1827,8 @@ impl Editor {
 
     pub fn ensure_cursor_in_view(&mut self, id: ViewId) {
         let config = self.config();
-        let view = self.tree.get_mut(id);
-        let doc = &self.documents[&view.doc];
+        let view = self.tree.get(id).clone();
+        let doc = self.document_mut(view.doc).unwrap();
```

Comment:
```suggestion
        let view = self.tree.get(id);
        let doc = doc_mut!(self, &view.doc);
```

Reply:
Fixed. Could you explain to me how this works though? That macro seems to expand to basically the same code I had, ending up with a mutable and immutable reference to editor through doc and view respectively, why doesn't that fail the borrow checker?

Reply:
https://github.com/helix-editor/helix/blob/109f53fb60f1b126c4c9afd97dd75972725d04ac/helix-view/src/macros.rs#L1-L8 has a good blurb about it: when you use a function like `Editor::document_mut` you're taking a mutable reference of the whole Editor while `doc_mut!` takes a partial borrow of just the `documents` field. And the borrow checker will let you take multiple mutable partial borrows from the same struct as long as it's not two mutable references of the same field.

Ideally the function and macros would be equivalent but the borrow checker isn't smart enough for that yet

Reply:
Thank you! That makes a lot of sense